modules/issue_tracker/php/uploadhelper.class.inc

```diff
@@ -200,6 +199,18 @@ class UploadHelper
         return $mime_type;
     }
 
+    /**
+     * Create a sha1 string from file_content.
+     *
+     * @param string $file_path
+     *
+     * @return string
+     */
+    function generate_hash(string $file_path) : string
+    {
+        return sha1_file($file_path);
```

Comment:
sha1 is already broken. A stronger hash using [PHP's generic hash function](https://www.php.net/manual/en/function.hash.php) should be used.

Reply:
@driusan how about https://www.geeksforgeeks.org/php-hash_file-function/ because it handles hashing the file? I want to hash from the file content and I believe hash_file() does that while hash() takes a string.

Reply:
as long as it's a built in function, it doesn't matter, they're the same thing.. the important thing is that a better hash than sha1 is used.

Reply:
@driusan latest commit uses md5 and works from quickly testing it.

Reply:
`md5` isn't great either. I'd suggest using at least sha256

Reply:
thanks @johnsaigle